contracts/BridgeTokens.sol

```diff
@@ -1,41 +1,118 @@
-    /// @param tokenId The ID of the token
-    /// @return The address of the token
-    function getAddress(uint8 tokenId) external view override returns (address) {
-        return supportedTokens[tokenId];
+    function getAddress(uint8 tokenId) public view override returns (address) {
+        return supportedTokens[tokenId].tokenAddress;
+    }
+
+    function getSuiDecimal(uint8 tokenId) public view override returns (uint8) {
+        return supportedTokens[tokenId].suiDecimal;
+    }
+
+    function isTokenSupported(uint8 tokenId) public view override returns (bool) {
+        return supportedTokens[tokenId].tokenAddress != address(0);
+    }
+
+    function convertEthToSuiDecimal(uint8 tokenId, uint256 amount)
+        public
+        view
+        override
+        tokenSupported(tokenId)
+        returns (uint64)
+    {
+        uint8 ethDecimal = IERC20Metadata(getAddress(tokenId)).decimals();
+        uint8 suiDecimal = getSuiDecimal(tokenId);
+
+        if (ethDecimal == suiDecimal) {
+            // Ensure the converted amount fits within uint64
+            require(amount <= type(uint64).max, "BridgeTokens: Amount too large for uint64");
+            return uint64(amount);
+        }
+
+        if (ethDecimal > suiDecimal) {
+            // Difference in decimal places
+            uint256 factor = 10 ** (ethDecimal - suiDecimal);
+            amount = amount / factor;
+        } else {
+            // Difference in decimal places
+            uint256 factor = 10 ** (suiDecimal - ethDecimal);
+            amount = amount * factor;
+        }
+
+        // Ensure the converted amount fits within uint64
+        require(amount <= type(uint64).max, "BridgeTokens: Amount too large for uint64");
+
+        return uint64(amount);
+    }
+
+    function convertSuiToEthDecimal(uint8 tokenId, uint64 amount)
+        public
+        view
+        override
+        tokenSupported(tokenId)
+        returns (uint256)
+    {
+        uint8 ethDecimal = IERC20Metadata(getAddress(tokenId)).decimals();
+        uint8 suiDecimal = getSuiDecimal(tokenId);
+
+        if (suiDecimal == ethDecimal) {
+            return uint256(amount);
+        }
+
+        uint256 convertedAmount;
+        if (ethDecimal > suiDecimal) {
+            // Difference in decimal places
+            uint256 factor = 10 ** (ethDecimal - suiDecimal);
+            convertedAmount = amount * factor;
+        } else {
+            // Difference in decimal places
+            uint256 factor = 10 ** (suiDecimal - ethDecimal);
+            convertedAmount = amount / factor;
+        }
+
+        return convertedAmount;
     }
 
     /// @dev Add a new supported token
     /// @param tokenId The ID of the token
     /// @param tokenAddress The address of the token
-    function addToken(uint8 tokenId, address tokenAddress) external onlyOwner {
-        supportedTokens[tokenId] = tokenAddress;
+    function updateToken(uint8 tokenId, address tokenAddress, uint8 suiDecimal)
```

Comment:
My though was we could add an "UpdateTokenSupport" message in the future. This function could be added to the `SuiBridge` contract as it is the owner of this `BridgeTokens` contract. Thoughts?

Reply:
I'm not sure we will do that before bridge v2. Is it better to leave this for v2 code?

Reply:
We can remove it to reduce complexity... I added it because we'll inevitably want to support new tokens and that means we'll need to update the `supportedTokens` mapping.  What we can do instead is deploy a whole new `BridgeToken` contract (BridgeTokenV2) that has these functions and then in the SuiBridgeV2 initializer we can update the referenced BridgeToken contract. Either seem like good options to me.

Reply:
let's do that since these will be unused functions until then!